src/metabase/channel/impl/email.clj

```diff
@@ -55,64 +171,111 @@
-                                                 (= :bcc recipient-type)
-                                                 (email/bcc-enabled?))}))
-
 (mu/defmethod channel/render-notification [:channel/email :notification/alert] :- [:sequential EmailMessage]
-  [_channel-type {:keys [card pulse payload pulse-channel]} _template recipients]
-  (let [condition-kwd             (messages/pulse->alert-condition-kwd pulse)
-        email-subject             (case condition-kwd
-                                    :meets (trs "Alert: {0} has reached its goal" (:name card))
-                                    :below (trs "Alert: {0} has gone below its goal" (:name card))
-                                    :rows  (trs "Alert: {0} has results" (:name card)))
-        {:keys [user-emails
-                non-user-emails]} (recipients->emails recipients)
-        timezone                  (channel.shared/defaulted-timezone card)
-        goal                      (find-goal-value card)
-        email-to-users            (when (> (count user-emails) 0)
-                                    (construct-email
-                                     email-subject user-emails
-                                     (messages/render-alert-email timezone pulse pulse-channel
-                                                                  [payload]
-                                                                  goal
-                                                                  nil)))
-        email-to-nonusers         (for [non-user-email non-user-emails]
-                                    (construct-email
-                                     email-subject [non-user-email]
-                                     (messages/render-alert-email timezone pulse pulse-channel
-                                                                  [payload]
-                                                                  goal
-                                                                  non-user-email)))]
-    (filter some? (conj email-to-nonusers email-to-users))))
+  [_channel-type {:keys [payload] :as notification-payload} template recipients]
+  (let [{:keys [card_part
+                alert
+                card]}     payload
+        timezone           (channel.shared/defaulted-timezone card)
+        rendered-card      (render-part timezone card_part {:pulse/include-title? true})
+        icon-attachment    (apply make-message-attachment (icon-bundle :bell))
+        attachments        (concat [icon-attachment]
+                                   (email-attachment rendered-card
+                                                     (assoc-attachment-booleans [alert] [card_part])))
+        message-context-fn (fn [non-user-email]
+                             (assoc notification-payload
+                                    :computed {:subject         (case (messages/pulse->alert-condition-kwd alert)
```

Comment:
computed here are the things that are generated by the channel itself.

Ideally we shouldn't do this and it should be taken care of by the template, but I don't think it's possible for things like `alert_content` which is quite complex to build here.